src/overworld.c

```diff
@@ -2975,102 +3037,116 @@ static u8 GetLinkPlayerIdAt(s16 x, s16 y)
+static bool8 FacingHandler_DoNothing(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj, u8 a3)
 {
-    return 0;
+    return FALSE;
 }
 
-static u8 sub_8087A20(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj, u8 a3)
+static bool8 FacingHandler_DpadMovement(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj, u8 a3)
 {
     s16 x, y;
 
-    eventObj->range.as_byte = npc_something3(a3, eventObj->range.as_byte);
+    eventObj->range.as_byte = FlipVerticalAndClearForced(a3, eventObj->range.as_byte);
     EventObjectMoveDestCoords(eventObj, eventObj->range.as_byte, &x, &y);
 
     if (LinkPlayerDetectCollision(linkPlayerEventObj->eventObjId, eventObj->range.as_byte, x, y))
     {
-        return 0;
+        return FALSE;
     }
     else
     {
         eventObj->directionSequenceIndex = 16;
         ShiftEventObjectCoords(eventObj, x, y);
         EventObjectUpdateZCoord(eventObj);
-        return 1;
+        return TRUE;
     }
 }
 
-static u8 sub_8087A88(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj, u8 a3)
+static bool8 FacingHandler_ForcedFacingChange(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj, u8 a3)
 {
-    eventObj->range.as_byte = npc_something3(a3, eventObj->range.as_byte);
-    return 0;
+    eventObj->range.as_byte = FlipVerticalAndClearForced(a3, eventObj->range.as_byte);
+    return FALSE;
 }
 
-static void sub_8087AA0(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj)
+// This is called every time a free movement happens. Most of the time it's a No-Op.
+static void MovementStatusHandler_EnterFreeMode(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj)
 {
-    linkPlayerEventObj->mode = 0;
+    linkPlayerEventObj->movementMode = MOVEMENT_MODE_FREE;
 }
 
-static void sub_8087AA8(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj)
+static void MovementStatusHandler_TryAdvanceScript(struct LinkPlayerEventObject *linkPlayerEventObj, struct EventObject *eventObj)
 {
     eventObj->directionSequenceIndex--;
-    linkPlayerEventObj->mode = 1;
+    linkPlayerEventObj->movementMode = MOVEMENT_MODE_FROZEN;
     MoveCoords(eventObj->range.as_byte, &eventObj->initialCoords.x, &eventObj->initialCoords.y);
     if (!eventObj->directionSequenceIndex)
     {
         ShiftStillEventObjectCoords(eventObj);
-        linkPlayerEventObj->mode = 2;
+        linkPlayerEventObj->movementMode = MOVEMENT_MODE_SCRIPTED;
     }
 }
 
-static u8 npc_something3(u8 a1, u8 a2)
+// Flip Up/Down facing codes. If newFacing doesn't specify a direction, default
+// to oldFacing. Note that this clears also the "FORCED" part of the facing code,
+// even for Left/Right codes.
+static u8 FlipVerticalAndClearForced(u8 newFacing, u8 oldFacing)
 {
-    switch (a1 - 1)
+    switch (newFacing - 1)
```

Comment:
Clean up the switch by removing all the -1s

Reply:
But if the -1's are removed, the constants can't be used. The switch operates by subtracting one from everything, shouldn't the cases reflect that?